### ISSUE TITLE
`return setmetatable(...)` causes false `call-non-callable`

### ISSUE THREAD

**Issue by lewis6991** [EmmyLuaLs/emmylua-analyzer-rust#1240]
Version: `emmylua_check 0.25.1`, default configuration.

`callable.lua`:

```lua
---@class Callable
---@overload fun(value: string): string
local M = {}

return setmetatable(M, {
  __call = function(_, value)
    return value
  end,
})
```

`main.lua`:

```lua
local callable = require('callable')
assert(callable('x') == 'x')
```

Running `emmylua_check . --warnings-as-errors` reports:

```text
main.lua:2:8
Cannot call expression of type `{ __call = fun(_: any, value: any) -> any ... }`. [call-non-callable]
```

Expected: no diagnostic. Calling `setmetatable(M, ...)` separately and then `return M` makes the check pass.

**Comment by xuhuanzy** [comment]
Since CppCXY is currently refactoring the core system, this issue will be addressed once the refactoring is completed or abandoned.
